source/common/config/datasource.cc

```diff
@@ -32,6 +37,46 @@ absl::optional<std::string> getPath(const envoy::config::core::v3::DataSource& s
              : absl::nullopt;
 }
 
+RemoteAsyncDataProvider::RemoteAsyncDataProvider(
+    Upstream::ClusterManager& cm, Init::Manager& manager,
+    const envoy::config::core::v3::RemoteDataSource& source, Event::Dispatcher& dispatcher,
+    Runtime::RandomGenerator& random, bool allow_empty, AsyncDataSourceCb&& callback)
+    : allow_empty_(allow_empty), callback_(std::move(callback)),
+      fetcher_(std::make_unique<Config::DataFetcher::RemoteDataFetcher>(cm, source.http_uri(),
+                                                                        source.sha256(), *this)),
+      init_target_("RemoteAsyncDataProvider", [this]() { start(); }),
+      retries_remaining_(
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(source.retry_policy(), num_retries, RetryCount)) {
+
+  uint64_t base_interval_ms = RetryInitialDelayMilliseconds;
+  uint64_t max_interval_ms = RetryMaxDelayMilliseconds;
+  if (source.has_retry_policy()) {
+    if (source.retry_policy().has_retry_back_off()) {
+      base_interval_ms =
+          PROTOBUF_GET_MS_REQUIRED(source.retry_policy().retry_back_off(), base_interval);
+      if (base_interval_ms < 1) {
+        base_interval_ms = 1;
+      }
```

Comment:
Should not be possible with schema check.

Reply:
I added a test to verify that. It seems that the schema check doesn't work here. I am not sure if I am missing sth.